### ISSUE TITLE
Public-api-checker doesn't properly understand implicit constructors.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#7574]
If i have ```public class C { }``` in v1 of roslyn and i change it to ```public class C { private C() { } }``` then this is not caught by the public-api checker.

It also improperly flags ```public class C { }``` to ```public class C { public C() { } }``` improperly as the addition of a public API.

**Comment by gafter** [comment]
@jaredpar Is there a separate repo for the API checker?

**Comment by jaredpar** [comment]
Yes it lives in https://github.com/dotnet/roslyn-analyzers

**Comment by jaredpar** [comment]
Migrated to the analyzer repo

https://github.com/dotnet/roslyn-analyzers/issues/806
